solaris/Cargo.toml

```diff
@@ -4,12 +4,14 @@ version = "0.1.0"
 authors = ["Tomasz Drwięga <tomusdrw@gmail.com>"]
 
 [dependencies]
-ethabi = { git = "https://github.com/paritytech/ethabi.git" }
-ethcore = { git = "https://github.com/paritytech/parity.git" }
-ethcore-transaction = { git = "https://github.com/paritytech/parity.git" }
-ethcore-bytes = { git = "https://github.com/paritytech/parity.git" }
-ethereum-types = "0.3"
-evm = { git = "https://github.com/paritytech/parity.git" }
+ethabi = { git = "https://github.com/openethereum/ethabi.git" }
+ethcore = { git = "https://github.com/openethereum/openethereum.git", branch = "stable", features = ["test-helpers"]}
+common-types = { git = "https://github.com/openethereum/openethereum.git", branch = "stable" }
+trace = { git = "https://github.com/openethereum/openethereum.git", branch = "stable" }
+spec = { git = "https://github.com/openethereum/openethereum.git", branch = "stable" }
```

Comment:
hi guys, so far the branch for the upcoming release will be based from master, is not there yet until last changes are implemented.

I guess if you want to try things out that should be  in another branch? or there is something about rust crates that I’m missing? so far stable branch is not being used as far as I know @vorot93

Reply:
The problem here is that we have here a project that is external to `openethereum` but uses code from it, fetching individual subcrates. It will break a lot. We can solve/improve on this in several ways, all of which have problems:

1. do nothing; dependent projects use git dependencies and break often and we just deal with that
1. pin dependencies to a single branch or revision, much like @606simps did here
1. publish the crates
1. bring this project or others into the `openethereum` repo

I think the second option is perhaps the least worst. And maybe for the time being we should pin it to a specific "known-good" revision, i.e. current master.

wdyt?

Reply:
I see.

Yeah, option 2 might be the better here. Why use a branch though? wouldn't be better a specific tag? I can see in the docs https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-git-repositories that tag versioning is possible, wouldn't not work well in this case?

Reply:
Yes tag works fine. I tried tag "v3.0.0-alpha.1", it doesn't compile. Which tag should I use?

Reply:
mmm it should compile, we have a CI for it and some releases https://github.com/openethereum/openethereum/actions/runs/68094801

What error are you getting? @606simps

Reply:
@dvdplm suggests to use `ethabi` version "12.0". I am getting a version conflict error.
```
    Updating git repository `https://github.com/openethereum/openethereum.git`
    Updating crates.io index
error: failed to select a version for `cc`.
    ... required by package `secp256k1-sys v0.1.1`
    ... which is depended on by `secp256k1 v0.17.2`
    ... which is depended on by `parity-crypto v0.5.0`
    ... which is depended on by `authority-round v0.1.0 (https://github.com/openethereum/openethereum.git?tag=v3.0.0-alpha.1#d4b5720d)`
    ... which is depended on by `spec v0.1.0 (https://github.com/openethereum/openethereum.git?tag=v3.0.0-alpha.1#d4b5720d)`
    ... which is depended on by `ethcore v1.12.0 (https://github.com/openethereum/openethereum.git?tag=v3.0.0-alpha.1#d4b5720d)`
    ... which is depended on by `solaris v0.1.0 (/home/stephen/git/Cartesi/sol-rs/solaris)`
    ... which is depended on by `solaris-example v0.1.0 (/home/stephen/git/Cartesi/sol-rs/example)`
versions that meet the requirements `>= 1.0.28, <= 1.0.41` are: 1.0.41, 1.0.40, 1.0.39, 1.0.38, 1.0.37, 1.0.36, 1.0.35, 1.0.34, 1.0.33, 1.0.32, 1.0.31, 1.0.30, 1.0.29, 1.0.28

all possible versions conflict with previously selected packages.

  previously selected package `cc v1.0.50`
    ... which is depended on by `backtrace-sys v0.1.35`
    ... which is depended on by `backtrace v0.3.46`
    ... which is depended on by `error-chain v0.12.2`
    ... which is depended on by `ethabi v9.0.1`
    ... which is depended on by `authority-round v0.1.0 (https://github.com/openethereum/openethereum.git?tag=v3.0.0-alpha.1#d4b5720d)`
    ... which is depended on by `spec v0.1.0 (https://github.com/openethereum/openethereum.git?tag=v3.0.0-alpha.1#d4b5720d)`
    ... which is depended on by `ethcore v1.12.0 (https://github.com/openethereum/openethereum.git?tag=v3.0.0-alpha.1#d4b5720d)`
    ... which is depended on by `solaris v0.1.0 (/home/stephen/git/Cartesi/sol-rs/solaris)`
    ... which is depended on by `solaris-example v0.1.0 (/home/stephen/git/Cartesi/sol-rs/example)`

failed to select a version for `cc` which could resolve this conflict
```

Reply:
```Cargo.toml
[dependencies]
ethabi = "12.0.0"
ethcore = { git = "https://github.com/openethereum/openethereum.git", tag = "v3.0.0-alpha.1", features = ["test-helpers"]}
common-types = { git = "https://github.com/openethereum/openethereum.git", tag = "v3.0.0-alpha.1" }
trace = { git = "https://github.com/openethereum/openethereum.git", tag = "v3.0.0-alpha.1" }
spec = { git = "https://github.com/openethereum/openethereum.git", tag = "v3.0.0-alpha.1" }
parity-bytes = "0.1"
ethereum-types = "0.9.0"
evm = { git = "https://github.com/openethereum/openethereum.git", tag = "v3.0.0-alpha.1" }
lazy_static = "0.2"
vm = { git = "https://github.com/openethereum/openethereum.git", tag = "v3.0.0-alpha.1" }
error-chain = "0.12.0"
```

Reply:
> previously selected package `cc v1.0.50`

So this is something that pops up sometimes. During dependency resolution some crate picked `cc` at `1.0.50` which causes `secp256k1` to fail. We've pinged upstream about their strict requirement on `cc` but for the time being your best bet is to manually edit `Cargo.lock`: find the dependency that picks `cc 1.0.50` and change it to `1.0.41` instead.

Reply:
`v3.0.0-alpha.1` has other dependencies conflict. What other tag should I use?
https://github.com/openethereum/openethereum/blob/d4b5720df0572a3f79783b1c9e55c765e20e2475/ethcore/Cargo.toml#L26
https://github.com/openethereum/openethereum/blob/d4b5720df0572a3f79783b1c9e55c765e20e2475/ethcore/machine/Cargo.toml#L31